projects/observability/src/shared/dashboard/data/graphql/explorer-visualization/explorer-visualization-cartesian-data-source.model.ts

```diff
@@ -0,0 +1,28 @@
+import { TimeDuration } from '@hypertrace/common';
+import { Model } from '@hypertrace/hyperdash';
+import {
+  ExploreRequestState,
+  ExploreVisualizationRequest
+} from '../../../../components/explore-query-editor/explore-visualization-builder';
+import { ExploreCartesianDataSourceModel } from '../explore/explore-cartesian-data-source.model';
+
+@Model({
+  type: 'explorer-visualization-cartesian-data-source'
+})
+export class ExplorerVisualizationCartesianDataSourceModel extends ExploreCartesianDataSourceModel {
+  public request?: ExploreVisualizationRequest;
+
+  protected buildRequestState(interval?: TimeDuration | 'AUTO'): ExploreRequestState | undefined {
+    if (this.request?.series.length === 0 || this.request?.context === undefined) {
+      return undefined;
+    }
+
+    return {
+      series: this.request.series,
```

Comment:
So previously, from what I recall, the model read off the request stream producing new data whenever the request changed - saving us from having to rebuild the dashboard on every explorer query change.

This is why the previous implementation piped off `this.request.exploreQuery$` rather than reading the current state. Does the change, as implemented, still respond to changes in the explorer query as the user interacts with explorer? And if so, is the dashboard being rebuilt each time (maybe it always was?)?

Reply:
Yeah. this was confusing. I have updated this in the latest commit.

Reply:
ok, so table should be fine with getData emitting new data over time.